src/main/java/au/id/micolous/metrodroid/transit/podorozhnik/PodorozhnikTransitData.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * PodorozhnikTransitData.java
+ *
+ * Copyright 2015-2016 Michael Farrell <micolous+git@gmail.com>
+ * Copyright 2018 Google
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package au.id.micolous.metrodroid.transit.podorozhnik;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.Nullable;
+import android.text.Spanned;
+
+import au.id.micolous.metrodroid.card.classic.ClassicCard;
+import au.id.micolous.metrodroid.card.classic.ClassicSector;
+import au.id.micolous.metrodroid.transit.TransitData;
+import au.id.micolous.metrodroid.transit.TransitIdentity;
+import au.id.micolous.metrodroid.util.Utils;
+
+import java.util.Arrays;
+
+/**
+ * Podorozhnik cards.
+ */
+
+public class PodorozhnikTransitData extends TransitData {
+
+    public static final String NAME = "Podorozhnik";
+    public static final String LONG_NAME = "Podorozhnik transit card";
+    public static final Parcelable.Creator<PodorozhnikTransitData> CREATOR = new Parcelable.Creator<PodorozhnikTransitData>() {
+        public PodorozhnikTransitData createFromParcel(Parcel parcel) {
+            return new PodorozhnikTransitData(parcel);
+        }
+
+        public PodorozhnikTransitData[] newArray(int size) {
+            return new PodorozhnikTransitData[size];
+        }
+    };
+
+    private static final String TAG = "PodorozhnikTransitData";
+
+    private int mBalance;
+
+    @Nullable
+    @Override
+    public Integer getBalance() {
+        return Integer.valueOf(mBalance);
+    }
+
+    @Override
+    public Spanned formatCurrencyString(int amount, boolean isBalance) {
+        return Utils.formatCurrencyString(amount, isBalance, "RUB");
+    }
+
+    @Override
+    public String getSerialNumber() {
+        return null;
+    }
+
+    private static int getBalance(ClassicSector sector) {
+        byte[] b = sector.getBlock(1).getData();
+        int sn = (b[0] & 0xff);
```

Comment:
Utils.reverseBuffer + Utils.byteArrayToBigInteger for little endian serial numbers.